### ISSUE TITLE
GetUserInfoAsync Newtonsoft.Json exception

### ISSUE THREAD

**Issue by oliverjanik** [auth0/auth0.net#390]
First time using auth0-net.

Nuget version: 7.0.4

This blew up on me today:
```csharp
var u = await auth0.GetUserInfoAsync(resp.AccessToken);
```

Stacktrace:
```
Newtonsoft.Json.JsonSerializationException: Error setting value to 'UpdatedAt' on 'Auth0.AuthenticationApi.Models.UserInfo'. ---> System.InvalidCastException: Specified cast is not valid.
   at SetUpdatedAt(Object , Object )
   at Newtonsoft.Json.Serialization.DynamicValueProvider.SetValue(Object target, Object value)
   --- End of inner exception stack trace ---
   at Newtonsoft.Json.Serialization.DynamicValueProvider.SetValue(Object target, Object value)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.SetPropertyValue(JsonProperty property, JsonConverter propertyConverter, JsonContainerContract containerContract, JsonProperty containerProperty, JsonReader reader, Object target)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateObject(Object newObject, JsonReader reader, JsonObjectContract contract, JsonProperty member, String id)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateObject(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.CreateValueInternal(JsonReader reader, Type objectType, JsonContract contract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerMember, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonSerializerSettings settings)
   at Newtonsoft.Json.JsonConvert.DeserializeObject[T](String value, JsonSerializerSettings settings)
   at Auth0.AuthenticationApi.HttpClientAuthenticationConnection.<SendRequest>d__8`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Auth0.AuthenticationApi.HttpClientAuthenticationConnection.<GetAsync>d__6`1.MoveNext()
```

Here's my assessment:
1. You're de-serializing without specifing `JsonSerializerSettings` over here: https://github.com/auth0/auth0.net/blob/release-7.0.4/src/Auth0.AuthenticationApi/HttpClientAuthenticationConnection.cs#L92
2. Which means default settings kick in `JsonConvert.DefaultSettings`. These are configured outside of this library.
3. You haven't tested all permutations of (mis)configurations that might be affecting your library. The encapsulation is broken, global settings are your dependency that you can't control.
4. In my case I suspect it's one of these
```
DateFormatHandling = DateFormatHandling.IsoDateFormat,
DateParseHandling = DateParseHandling.None,
DateTimeZoneHandling = DateTimeZoneHandling.Unspecified,
```
5. It's not feasible for me to modify my global `JsonSerializerSettings` which my entire solution depends on.

Newtonsoft.Json and date/time don't mix well, this library has caused me much hairloss in the past. Do not underestimate it. I hope you can drop it for .net core 3+

Are you able to look into this? I'll do a bit more prodding around.

**Comment by oliverjanik** [comment]
Alright here's a straight forward way to break your tests:

https://github.com/oliverjanik/auth0.net/commit/66d5acd27a56903fe9c485a821279692fbfcbca2

The solution here seems to be to provide your own deserialisation settings.

**Comment by bradwilson** [comment]
Having static global defaults is a bad idea (regardless of library), and pushing on those static global defaults is going to break you with a lot of stuff, not just this library. I'd strongly recommend you, maybe don't? For your own sanity. 😉 

(Outside observer's opinion)

**Comment by damieng** [comment]
I have a PR open to fix this as I want to make sure we're not broken by global settings for JSON.Net.

I do think however by changing those global settings you're probably going to run into similar issues with other libraries that use JSON.Net.

System.Text.Json would be nice to adopt but it does mean dropping versions of .NET Framework we currently support (4.5.2) which would need to be scheduled for a new major release.

**Comment by oliverjanik** [comment]
Your library is the first for me, thanks for fixing. Those global settings exist so that I can modify the to my hearts content. `DateParseHandling` default is a particularly dumb setting that is non standards compliant.

Regardless, users of your library might not have a choice or say in global settings imposed on them so thanks for fixing that.

My preference would be to completely dump the bloated Newtonsoft.Json.

**Comment by damieng** [comment]
Shipped in 7.0.5 now available on NuGet.

**Comment by oliverjanik** [comment]
I can still replicate with 7.0.5 was the fix shipped?

**Comment by damieng** [comment]
It appears that the JsonSerializerSettings class although it looks like a plain object actually keeps track of which properties were set internally and JsonConvert.DefaultSettings gets called first and then the JsonSerializerSettings you supply gets applied over the top - only for the settings that you specifically set.

**Comment by oliverjanik** [comment]
The plot thickens, can you reopen please?

**Comment by damieng** [comment]
There is only so much we can do to avoid ill-advised global configuration changes on system-wide components. 

We're going to manually specify DateParseHandling = DateParseHandling.DateTime to force the default setting back for JSON dates inside Auth0.NET but any further global configurations either on JSON.NET or other shared .NET components would be out of scope.

**Comment by damieng** [comment]
Now shipped in 7.0.6

**Comment by oliverjanik** [comment]
Thank you very much. The real solution here would be to drop the obnoxious dependency.
